python/foglamp/common/microservice_management_client/microservice_management_client.py

```diff
@@ -166,3 +170,115 @@ def get_services(self, service_name=None, service_type=None):
+            _logger.error("Client error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        if r.status in range(500, 600):
+            _logger.error("Server error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        res = r.read().decode()
+        self._management_client_conn.close()
+        response = json.loads(res)
+        return response
+
+    def create_configuration_category(self, category_data):
+        """
+
+        :param category_data: e.g. '{"key": "TEST", "description": "description", "value": {"info": {"description": "Test", "type": "boolean", "default": "true"}}}'
+        :return:
+        """
+        url = '/foglamp/service/category'
+
+        self._management_client_conn.request(method='POST', url=url, body=category_data)
+        r = self._management_client_conn.getresponse()
+        if r.status in range(400, 500):
+            _logger.error("Client error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        if r.status in range(500, 600):
+            _logger.error("Server error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        res = r.read().decode()
+        self._management_client_conn.close()
+        response = json.loads(res)
+        return response
+
+    def update_configuration_item(self, category_name, config_item, category_data):
+        """
+
+        :param category_name:
+        :param config_item:
+        :param category_data: e.g. '{"value": "true"}'
+        :return:
+        """
+        url = "/foglamp/service/category/{}/{}".format(category_name, config_item)
+
+        self._management_client_conn.request(method='PUT', url=url, body=category_data)
+        r = self._management_client_conn.getresponse()
+        if r.status in range(400, 500):
+            _logger.error("Client error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        if r.status in range(500, 600):
+            _logger.error("Server error code: %d, Reason: %s", r.status, r.reason)
+            raise client_exceptions.MicroserviceManagementClientError(status=r.status, reason=r.reason)
+        res = r.read().decode()
+        self._management_client_conn.close()
+        response = json.loads(res)
+        return response
+
+    def delete_configuration_item(self, category_name, config_item):
+        """
+
+        :param category_name:
+        :param config_item:
+        :return:
+        """
+        url = "/foglamp/service/category/{}/{}".format(category_name, config_item)
```

Comment:
let's keep the route URL consistent with Config Admin Api

Reply:
done